src/librustc_mir/interpret/operator.rs

```diff
@@ -266,6 +203,9 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             (Div, F64) => f64_arithmetic!(/, l, r),
             (Rem, F64) => f64_arithmetic!(%, l, r),
 
+            (Eq, _) => PrimVal::from_bool(l == r),
+            (Ne, _) => PrimVal::from_bool(l != r),
```

Comment:
I see. ptr_op still pretty much has to support them (well, at least in its type), but I guess for the purpose of CTFE it makes sense to split this.